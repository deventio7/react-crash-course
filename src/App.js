import React, { Component } from 'react';
import SampleChildComponent from './SampleChildComponent';
import PureChildComponent from './PureChildComponent';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      componentText: '1'
    }
  };

  componentTextSetter = (newText) => {
    this.setState({
      componentText: newText
    });
  };

  render = () => {
    console.log('Rendering App');
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <div className="App-body">
          <SampleChildComponent pureComponentTextSetter={ this.componentTextSetter }/>
          <PureChildComponent passedInt={ this.state.componentText }/>
        </div>
      </div>
    );
  }
}

export default App;
