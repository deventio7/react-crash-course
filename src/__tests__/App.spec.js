import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

describe('App', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<App />);
  });

  it('render shallow matches snap', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('componentTextSetter changes state', () => {
    const wrapper = shallow(<App />);
    jest.spyOn(wrapper.instance(), 'setState');
    wrapper.instance().componentTextSetter('TEST VALUE');
    expect(wrapper.instance().setState).toBeCalledWith({ componentText: 'TEST VALUE' });
    expect(wrapper.instance().state.componentText).toBe('TEST VALUE');
  });
})
