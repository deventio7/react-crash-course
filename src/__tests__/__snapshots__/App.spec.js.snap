// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render shallow matches snap 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <div
    className="App-body"
  >
    <Connect(SampleChildComponent)
      pureComponentTextSetter={[Function]}
    />
    <PureChildComponent
      passedText="uninitialized text"
    />
  </div>
</div>
`;
